### ISSUE TITLE
unable to load mxnet library in R

### ISSUE THREAD

**Issue by wittyalias** [apache/incubator-mxnet#14004]
Possibly related to the issues in [#13952](https://github.com/apache/incubator-mxnet/pull/13952). 

I've followed the [instructions ](https://mxnet.incubator.apache.org/versions/master/install/index.html?platform=MacOS&language=R&processor=CPU) to install CPU version of MXNet for R on MacOS (10.14.2). 
I successfully install openblas (0.3.5) and opencv (4.0.1) via brew using
```
brew install opencv
brew install openblas
```
Also, added the symbolic links as suggested (this is a different openblas version that I have installed):
```
ln -sf /usr/local/opt/openblas/lib/libopenblas.dylib /usr/local/opt/openblas/lib/libopenblasp-r0.3.1.dylib
```
And then installed the package within R with:
```
cran <- getOption("repos")
cran["dmlc"] <- "https://apache-mxnet.s3-accelerate.dualstack.amazonaws.com/R/CRAN/"
options(repos = cran)
install.packages("mxnet")
```

## Description
When I try to load the mxnet library I get an error message below

```
>library(mxnet)
Error: package or namespace load failed for ‘mxnet’:
 .onLoad failed in loadNamespace() for 'mxnet', details:
  call: dyn.load(file, DLLpath = DLLpath, ...)
  error: unable to load shared object '/Library/Frameworks/R.framework/Versions/3.5/Resources/library/mxnet/libs/libmxnet.so':
  dlopen(/Library/Frameworks/R.framework/Versions/3.5/Resources/library/mxnet/libs/libmxnet.so, 10): Library not loaded: /usr/local/opt/openblas/lib/libopenblasp-r0.3.1.dylib
  Referenced from: /Library/Frameworks/R.framework/Versions/3.5/Resources/library/mxnet/libs/libmxnet.so
  Reason: image not found
```

## Environment info (Required)
I'm using R in RStudio 1.1.463
```
R version 3.5.2 (2018-12-20)
Platform: x86_64-apple-darwin15.6.0 (64-bit)
Running under: macOS Mojave 10.14.2

Matrix products: default
BLAS: /System/Library/Frameworks/Accelerate.framework/Versions/A/Frameworks/vecLib.framework/Versions/A/libBLAS.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/3.5/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods  
[7] base     
```

## What have you tried to solve it?

1. My guess is that its looking for different versions of openblas and opencv than I have installed, so I tried to install a previous versions of them via brew, but I can't figure out how to do it for openblas and 3.4 opencv installs to a folder 'opencv@3' - so the library won't be able to find it anyway. 

I did not change the symbolic link because before I had the openblas 0.3.1 error, I had a very similar looking error that said it couldn't find opencv 3.4.

I was thinking of cloning the repository and seeing if I could change what openblas and opencv versions mxnet looks for, but that seems like a lot of work for something that you may already be looking into.

**Comment by frankfliu** [comment]
@mxnet-label-bot add [build, R]

**Comment by saberioon** [comment]
I have the same issue, Is there any solution available?

**Comment by drag05** [comment]
I have installed mxnet for CPU and worked fine on Windows 10, R version 3.6.2. Then I wanted to try mxnet for GPU: I installed mxnet for GPU using the command:

`install.packages("https://s3.ca-central-1.amazonaws.com/jeremiedb/share/mxnet/GPU/mxnet.zip", repos = NULL)`

everything went file. However when trying to load in R version 3.6.2 or R Studio, I get the following error:

`Loading required package: mxnet`

`Error: package or namespace load failed for ‘mxnet’:
 .onLoad failed in loadNamespace() for 'mxnet', details:
  call: inDL(x, as.logical(local), as.logical(now), ...)
  error: unable to load shared object 'C:/R/R-3.6.2/library/mxnet/libs/x64/libmxnet.dll':
  LoadLibrary failure:  The specified module could not be found.`

I have been using my GPU for R/keras/tensorflow having installed cuda100 on miniconda3. It works quite smoothly on NVIDIA Geforce RTX 2080

Please advise

**Comment by sbar365** [comment]
I am also trying to install mxnet for R on MacOS (10.15.4). I installed openblas 0.3.9 and opencv 4.2.0_3. 

cran <- getOption("repos")
cran["dmlc"] <- "https://apache-mxnet.s3-accelerate.dualstack.amazonaws.com/R/CRAN/"
options(repos = cran)
install.packages("mxnet")
library(mxnet)

and I get the following error: 

[1] "Loading local: inst/libs/libmxnet.so"
Error: package or namespace load failed for ‘mxnet’:
 .onLoad failed in loadNamespace() for 'mxnet', details:
  call: dyn.load("R-package/inst/libs/libmxnet.so", local = FALSE)
  error: unable to load shared object '/Users/Stefanos/R-package/inst/libs/libmxnet.so':
  dlopen(/Users/Stefanos/R-package/inst/libs/libmxnet.so, 10): image not found

> sessionInfo()
R version 3.5.2 (2018-12-20)
Platform: x86_64-apple-darwin15.6.0 (64-bit)
Running under: macOS  10.15.4

Matrix products: default
BLAS: /System/Library/Frameworks/Accelerate.framework/Versions/A/Frameworks/vecLib.framework/Versions/A/libBLAS.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/3.5/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] devtools_2.2.2 usethis_1.5.1 

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.4         tidyr_1.0.2        prettyunits_1.1.1  ps_1.3.2           visNetwork_2.0.9   assertthat_0.2.1  
 [7] rprojroot_1.3-2    digest_0.6.25      R6_2.4.1           backports_1.1.5    ggplot2_3.3.0      pillar_1.4.3      
[13] rlang_0.4.5        rstudioapi_0.11    callr_3.4.3        drat_0.1.5         DiagrammeR_0.9.2   desc_1.2.0        
[19] downloader_0.4     servr_0.16         readr_1.3.1        stringr_1.4.0      htmlwidgets_1.5.1  igraph_1.2.5      
[25] munsell_0.5.0      compiler_3.5.2     influenceR_0.1.0   rgexf_0.16.0       httpuv_1.5.2       xfun_0.12         
[31] pkgconfig_2.0.3    pkgbuild_1.0.6     htmltools_0.4.0    tidyselect_1.0.0   tibble_3.0.0       gridExtra_2.3     
[37] XML_3.99-0.3       fansi_0.4.1        viridisLite_0.3.0  crayon_1.3.4       dplyr_0.8.5        withr_2.1.2       
[43] later_1.0.0        grid_3.5.2         jsonlite_1.6.1     gtable_0.3.0       lifecycle_0.2.0    magrittr_1.5      
[49] scales_1.1.0       neuralnet_1.44.2   cli_2.0.2          stringi_1.4.6      viridis_0.5.1      fs_1.3.2          
[55] promises_1.1.0     remotes_2.1.1      testthat_2.3.2     ellipsis_0.3.0     vctrs_0.2.4        RColorBrewer_1.1-2
[61] tools_3.5.2        glue_1.3.2         purrr_0.3.3        hms_0.5.3          processx_3.4.2     pkgload_1.0.2     
[67] yaml_2.2.1         colorspace_1.4-1   sessioninfo_1.1.1  memoise_1.1.0 

Any ideas how to fix this issue?
Thank you.
